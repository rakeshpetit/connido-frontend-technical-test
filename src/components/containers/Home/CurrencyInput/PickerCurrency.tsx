import React, { Component } from 'react';
import { View, Picker, StyleSheet } from 'react-native';

type Props = {
    symbols: Array<any>,
    setDestinationCurrencyDone: Function,
    destinationCurrency: string
};

class PickerCurrency extends Component<Props, any> {
    constructor(props: any) {
        super(props);
        this.state = {
            currency: this.props.destinationCurrency
        };
    }
    onCurrencySelect = (itemValue: string) => {
        this.setState({currency: itemValue});
        this.props.setDestinationCurrencyDone(itemValue);
    }
  render() {
    if (this.props.symbols.length <= 0) {
        return undefined;
    }
    return (
        <View style={styles.container}>
            <Picker
                selectedValue={this.state.currency}
                style={{
                    backgroundColor: "white",
                    alignContent: "center",
                    height: 200,
                    width: 100
                }}
                itemStyle={{ backgroundColor: "white" }}
                onValueChange={(itemValue: string) =>
                    this.onCurrencySelect(itemValue)
                }>
                {this.props.symbols.map((symbol, index) => {
                    return (
                        <Picker.Item
                            key={index}
                            label={symbol.key}
                            value={symbol.key}
                        />
                    );
                })}
            </Picker>
        </View>
    );
  }
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: "white",
    }
});

export default PickerCurrency;
