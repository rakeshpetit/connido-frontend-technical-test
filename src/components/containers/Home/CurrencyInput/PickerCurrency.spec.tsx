import React from 'react';
import renderer from 'react-test-renderer';

import PickerCurrency from './PickerCurrency';

describe('Header', () => {
    it('renders nothing when there are no symbols', () => {
    const symbols = [{}];
    const setDestinationCurrencyDone = () => {};
    const destinationCurrency = 'GBP';
    const tree = renderer.create
        (<PickerCurrency 
            symbols={symbols}
            setDestinationCurrencyDone={setDestinationCurrencyDone}
            destinationCurrency={destinationCurrency}
        />).toJSON();
    expect(tree).toMatchSnapshot();
    });
    it('renders correctly when there is symbol data', () => {
        const symbols = [{key: 'USD', value: 'United States Dollar'}];
        const setDestinationCurrencyDone = () => {};
        const destinationCurrency = 'GBP';
        const tree = renderer.create
            (<PickerCurrency 
                symbols={symbols}
                setDestinationCurrencyDone={setDestinationCurrencyDone}
                destinationCurrency={destinationCurrency}
            />).toJSON();
        expect(tree).toMatchSnapshot();
        });
});