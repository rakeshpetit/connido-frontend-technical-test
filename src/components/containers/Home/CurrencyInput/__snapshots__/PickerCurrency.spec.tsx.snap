// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders PickerCurrency correctly when there is symbol data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "height": 50,
        "width": 100,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "USD",
            "textColor": undefined,
            "value": "USD",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Header renders correctly when there is symbol data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "height": 50,
        "width": 100,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "USD",
            "textColor": undefined,
            "value": "USD",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Header renders nothing when there are no symbols 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "height": 50,
        "width": 100,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": undefined,
            "textColor": undefined,
            "value": undefined,
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
